content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,330 @@
+that refer to or use them.
+
+* FAQ: [How does Flux run `kustomize build` internally?][Kustomize Build Flags] (Replicate Kustomize behavior locally with the Kustomize CLI)
+
+### 10. Kustomize Controller (Server Side Apply)
+
+**Needs a link to FluxCD documentation for Server-Side Apply process!** _The only current on-point reference is the [Server-side reconciliation is coming](https://fluxcd.io/blog/2021/09/server-side-reconciliation-is-coming/) blog post.  This would likely be considered a gap in FluxCD project docs._
+
+The Kustomize Controller communicates directly with the Kubernetes API using
+[server-side apply and update][] API operations
+instead of running the `kubectl apply` command as a separate forked process and passing it manifest data through
+a system pipe. Applying resource manifests directly
+to the Kubernetes API is both more efficient and provides more control over the process, enabling the Kustomize
+Controller to give real-time feedback on validation errors, garbage-collection and resource health assessment.
+It also allows the Kubernetes API to track
+[field management][],
+so different management tools or controllers can set field values within the same resource without interfering
+with each other.
+
+The server-side apply operation is synchronous rather than asynchronous. If any resources fail to become ready before a
+specified timeout, the controller can abort the entire transaction.
+
+The Kustomize Controller applies resource manifests to match the order in which they were rendered by the kustomize `build`
+call.  It therefore applies any custom resource definition (CRD), namespace, or cluster-scoped resources before their subordinate
+custom resource or namespace-scoped resources to that they will be available in the API for the resources that refer to or use them.
+
+<insert sequence diagram>?
+
+### 11. Helm Controller (`HelmRelease` Custom Resource)
+
+A [HelmRelease][HelmRelease API] is a composition of a chart, the chart values (parameters to the chart), and any inputs like secrets or config maps that are used to compose the values. Declaring a HelmRelease will cause the Helm Controller to perform an install using the Helm client libraries. Any changes made to the HelmRelease will trigger the Helm Controller to perform an upgrade to actuate those changes. You can find more information about HelmReleases [here][HelmRelease Guide] and more general info about Helm for Flux users [here][Helm Use Case].
+
+
+### 12. HelmRepository and HelmChart (Sources for Helm)
+
+A Helm Repository is the native and preferred source for Helm. The Helm Controller works in tandem with the Source Controller to provide a [HelmRepository API][] that collects the correct release version from the helm repo and republishes its data as a [HelmChart][HelmChart API] artifact (another .tar.gz).
+
+The helm repo itself is represented internally in the Source Controller as a YAML index of all releases, including any charts in the repository.
```

Comment:
Might be helpful to explain that users never explicitly create a `HelmChart` and how the definition of a `HelmRelease` is what the SC uses to create a `HelmChart`.